### PR TITLE
fix: leader election create cm in target namespace

### LINKED ISSUE #2437: leaderelection: error initially creating leader election record
## Bug Report

**What version of Chaos Mesh are you using?**
master branch, commit https://github.com/chaos-mesh/chaos-mesh/commit/a5589134dd2a9735cecd627d34c55582f3dc6aee
**What did you do?**

install Chaos Mesh with `clusterScoped=false controllerManager.targetNamespace=busybox `

**What did you expect to see?**

Chaos Mesh run success.

**What did you see instead?**

see some error in controller-manager's log:

```log
leaderelection.go:329] error initially creating leader election record: configmaps is forbidden: User "system:serviceaccount:chaos-testing:chaos-controller-manager" cannot create resource "configmaps" in API group "" in the namespace "chaos-testing"

```

### PR DESCRIPTION
**Pull request by iguoyr**
Signed-off-by: SiyuChen <ryougi201@gmail.com>

### What problem does this PR solve?

Close #2437

> Problem Summary:

### What's changed and how it works?

> What's Changed: leader election will create ConfigMap in targetNamespace instead of namespace that chaos mesh installed.

### Related changes

- [ ] Need to update `chaos-mesh/website`
- [ ] Need to update `Chaos Dashboard`
- [ ] Need to **cheery-pick to release branches**

### Checklist

Tests

- [ ] Unit test
- [ ] E2E test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Breaking backward compatibility

### Release note 

```text
Please add a release note.

You can safely ignore this section if you don't think this PR needs a release note.
```

### CODE REVIEW

**Comment by iguoyr** [comment]
/cc @WangXiangUSTC 
PTAL

**Comment by STRRL** [comment]
I think it's better to grant permission on configmap in chaos-mesh's namespace. Because leader election is the feature for controller-manager itself. It should not leak to other ns. Once the target namespace is deleted, chaos-mesh should work well.

**Comment by iguoyr** [comment]
> I think it's better to grant permission on configmap in chaos-mesh's namespace. Because leader election is the feature for controller-manager itself. It should not leak to other ns. Even the target namespace is deleted, chaos-mesh should work well.

@STRRL Thanks for the correction! Already fixed!

File: helm/chaos-mesh/templates/controller-manager-rbac.yaml [hunk 1]
```diff
@@ -57,9 +57,6 @@ rules:
       - watch
       - list
       - get
-  - apiGroups: [ "" ]
-    resources: [ "configmaps" ]
-    verbs: [ "*" ]
```
Comment (WangXiangUSTC) [inline helm/chaos-mesh/templates/controller-manager-rbac.yaml:62]: Is there any other problem with changing from `*` to `"get", "list", "watch", "delete", "update", "patch"`?

Comment (iguoyr) [reply helm/chaos-mesh/templates/controller-manager-rbac.yaml:62]: `create` is also required, so it's basically the same as '*'

**Review by WangXiangUSTC — APPROVED** [review]
LGTM

**Review by YangKeao — APPROVED** [review]
LGTM

**Pull request merged**